src/components/WalletSelector/GnosisSafePendingMessage.tsx

```diff
@@ -1,15 +1,71 @@
+import Button from 'components/core/Button/Button';
 import { BodyRegular, TitleMedium } from 'components/core/Text/Text';
 import Spacer from 'components/core/Spacer/Spacer';
 import { LISTENING_ONCHAIN, PendingState, PROMPT_SIGNATURE } from 'types/Wallet';
 import colors from 'components/core/colors';
 import styled, { keyframes } from 'styled-components';
+import { useMemo, useEffect } from 'react';
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers/lib/web3-provider';
+import { getLocalStorageItem } from 'utils/localStorage';
+
+const GNOSIS_NONCE_STORAGE_KEY = 'gallery_gnosis_nonce';
 
 type Props = {
   pendingState: PendingState;
   userFriendlyWalletName: string;
+  onRestartClick?: () => void;
+  manuallyValidateSignature: () => void;
```

Comment:
superhypernit: we should have a convention where function props are named with `on`; otherwise this could be confused for a boolean